actors/market/tests/on_miner_sectors_terminate.rs

```diff
@@ -43,28 +44,48 @@ fn terminate_multiple_deals_from_multiple_providers() {
         .collect::<Vec<DealID>>()
         .try_into()
         .unwrap();
-    activate_deals(&rt, sector_expiry, PROVIDER_ADDR, current_epoch, &[deal1, deal2, deal3]);
+    activate_deals_legacy(&rt, sector_expiry, PROVIDER_ADDR, current_epoch, &[deal1, deal2, deal3]);
 
     let addrs = MinerAddresses { provider: provider2, ..MinerAddresses::default() };
     let deal4 = generate_and_publish_deal(&rt, CLIENT_ADDR, &addrs, start_epoch, end_epoch);
     let deal5 = generate_and_publish_deal(&rt, CLIENT_ADDR, &addrs, start_epoch, end_epoch + 1);
-    activate_deals(&rt, sector_expiry, provider2, current_epoch, &[deal4, deal5]);
+    activate_deals_legacy(&rt, sector_expiry, provider2, current_epoch, &[deal4, deal5]);
 
-    terminate_deals(&rt, PROVIDER_ADDR, &[deal1]);
-    assert_deals_terminated(&rt, current_epoch, &[deal1]);
+    let prop1 = get_deal_proposal(&rt, deal1);
+    terminate_deals_and_assert_balances(&rt, CLIENT_ADDR, PROVIDER_ADDR, &[deal1]);
+    assert_deal_deleted(&rt, deal1, prop1);
     assert_deals_not_terminated(&rt, &[deal2, deal3, deal4, deal5]);
 
-    terminate_deals(&rt, provider2, &[deal5]);
-    assert_deals_terminated(&rt, current_epoch, &[deal5]);
+    let prop5 = get_deal_proposal(&rt, deal5);
+    terminate_deals_and_assert_balances(&rt, CLIENT_ADDR, provider2, &[deal5]);
+    assert_deal_deleted(&rt, deal5, prop5);
     assert_deals_not_terminated(&rt, &[deal2, deal3, deal4]);
 
-    terminate_deals(&rt, PROVIDER_ADDR, &[deal2, deal3]);
-    assert_deals_terminated(&rt, current_epoch, &[deal2, deal3]);
+    let prop2 = get_deal_proposal(&rt, deal2);
+    let prop3 = get_deal_proposal(&rt, deal3);
+    terminate_deals_and_assert_balances(&rt, CLIENT_ADDR, PROVIDER_ADDR, &[deal2, deal3]);
+    assert_deal_deleted(&rt, deal2, prop2);
+    assert_deal_deleted(&rt, deal3, prop3);
     assert_deals_not_terminated(&rt, &[deal4]);
 
-    terminate_deals(&rt, provider2, &[deal4]);
-    assert_deals_terminated(&rt, current_epoch, &[deal4]);
-    check_state(&rt);
+    let prop4 = get_deal_proposal(&rt, deal4);
+    terminate_deals_and_assert_balances(&rt, CLIENT_ADDR, provider2, &[deal4]);
+    assert_deal_deleted(&rt, deal4, prop4);
+    check_state_with_expected(
+        &rt,
+        &[
+            Regex::new("^deal op found for deal id \\d+ with missing proposal at epoch \\d+$")
```

Comment:
If this is to be a reachable state in the future (because neither sync termination nor activation remove the queue entry) we should instead remove this from the state checks.

Reply:
this _is_ a reachable state but only transiently. it should eventually be cleaned up in cron. I think it's still valuable as a state invariant check as these shouldn't dangle long term and we should check cron actually removes them.

Reply:
Some more context is that these state checks are ported to Go and run post-upgrade and at other times. They're not just for unit tests. A failed expectation will raise a big alarm bell if observed on mainnet. I can see that this should be only transient and in tests we want to ensure these get cleaned up. But maybe we should do that explicitly rather than as part of these invariants. FYI @ZenGround0